### PR TITLE
Endpoints for listing all segments/metrics :chart_with_upwards_trend:

### PR DESCRIPTION
**Pull request by camsaul**
-  Added new `GET /api/segment/` and `GET/api/metric` endpoints for @lewisl9029 
-  Added tests for new endpoints
-  Some minor cleanup

### CODE REVIEW

File: src/metabase/api/metric.clj [hunk 1]
```diff
@@ -25,6 +26,13 @@
   (check-superuser)
   (check-404 (metric/retrieve-metric id)))
 
+(defendpoint GET "/"
+  "Fetch all `Metrics`. You must be a superuser to do this."
+  [id]
+  (check-superuser)
```
Comment (salsakran) [inline src/metabase/api/metric.clj]: Why's this a superuser only? 

The data model reference is in userland, not the admin.

**Comment by salsakran** [comment]
other than the superuser check, looks solid to me.

File: src/metabase/api/metric.clj (hunk 1 above)
Comment (camsaul) [inline src/metabase/api/metric.clj]: All the other endpoints to get metrics & segments require superuser status so I was just going along with that ?

File: src/metabase/api/segment.clj [hunk 2]
```diff
@@ -25,6 +26,12 @@
   (check-superuser)
   (check-404 (segment/retrieve-segment id)))
 
+(defendpoint GET "/"
+  "Fetch *all* `Segments`."
+  []
+  (-> (db/select Segment, :is_active true)
+      (hydrate :creator)))
```
Comment (salsakran) [inline src/metabase/api/segment.clj]: Does this fetch all creators in a single SELECT or are we fetching each one separately?

Comment (camsaul) [inline src/metabase/api/segment.clj]: single select

Comment (salsakran) [inline src/metabase/api/segment.clj]: :metal:

**Comment by salsakran** [comment]
I'm 👍

**Comment by camsaul** [comment]
need to fix a couple of things give me a sec

**Pull request merged**